### DISCUSSION TITLE
Keycloak 19.0.2 client secret

### DISCUSSION THREAD

**Discussion by NajmZaga** [keycloak/keycloak#14724] [Q&A] [answered]
Hello keycloakers,

I recently downloaded Keycloak 19.0.2 and I'm having a hard time searching for the client secret.
I did my researches and found that the _"Credentials"_ tab is no longer available in the new UI.

Any idea where to find the client secret and is it deprecated or still working ?

Thank you in advance.

**Comment by souravs17031999** [comment, marked as answer]
I am able to get credentials tab in new UI.
Probably check which Oauth flow client is supporting.
Not all of them requires credentials and so if not , it will not be visible.
For ex. if you check "service accounts roles" or "direct access grants" which corresponds to client_credentials and resource owner password grant flows, then you can see credentials tab since these flows requires "client_secret".

![clients](https://user-images.githubusercontent.com/33771969/194713037-d295ec4e-3cd2-40fe-b9e0-7fa667be1b60.PNG)

**Reply by NajmZaga** [reply]
Thanks dude
